l1-contracts/src/core/Rollup.sol

```diff
@@ -123,14 +132,13 @@ contract Rollup is Leonidas, IRollup, ITestRollup {
       revert Errors.Rollup__NotReadyToPrune(currentSlot, prunableAtSlot);
     }
 
-    // @note  We are not deleting the blocks, but we are "winding back" the pendingBlockCount
-    //        to the last block that was proven.
-    //        The reason we can do this, is that any new block proposed will overwrite a previous block
-    //        so no values should "survive". It it is however slightly odd for people reading
-    //        the chain separately from the contract without using pendingBlockCount as a boundary.
-    pendingBlockCount = provenBlockCount;
+    // @note  We are not deleting the blocks, but we are "winding back" the pendingTip to the last block that was proven.
+    //        We can do because any new block proposed will overwrite a previous block in the block log,
+    //        so no values should "survive".
+    //        People must therefore read the chain using the pendingTip as a boundary.
+    state.pendingTip = state.provenTip;
 
-    emit PrunedPending(provenBlockCount, pendingBlockCount);
+    emit PrunedPending(state.pendingTip.blockNumber);
```

Comment:
Would emit both, right now you don't learn how far back you actually end up going.

Reply:
On this, the event actually say "proven" but you are using pending.